controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -2266,12 +2327,47 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
           lintIssues: [...lintIssues.warnings, ...lintIssues.errors],
         });
 
+        let graphPruningIssues: SchemaGraphPruningIssues = { warnings: [], errors: [] };
+        if (namespace.enableGraphPruning && opts.chClient && newGraphQLSchema) {
+          const graphPruningConfigs = await schemaGraphPruningRepo.getNamespaceGraphPruningConfig(namespace.id);
+          if (graphPruningConfigs.length > 0) {
+            const usageRepo = new UsageRepository(opts.chClient);
+            const schemaGraphPruner = new SchemaGraphPruner(fedGraphRepo, subgraphRepo, usageRepo, newGraphQLSchema);
+
+            graphPruningIssues = await schemaGraphPruner.schemaGraphPruneCheck({
+              subgraph,
+              graphPruningConfigs,
+              updatedFields: schemaChanges.changes.filter(
+                (change) =>
+                  change.changeType === 'FIELD_ADDED' ||
+                  change.changeType === 'FIELD_TYPE_CHANGED' ||
+                  change.changeType === 'INPUT_FIELD_ADDED' ||
+                  change.changeType === 'INPUT_FIELD_TYPE_CHANGED' ||
+                  change.changeType === 'FIELD_ARGUMENT_ADDED' ||
+                  change.changeType === 'FIELD_ARGUMENT_REMOVED' ||
+                  change.changeType === 'FIELD_DEPRECATION_ADDED',
+              ),
+              removedFields: schemaChanges.changes.filter(
+                (change) => change.changeType === 'FIELD_REMOVED' || change.changeType === 'INPUT_FIELD_REMOVED',
+              ),
+              organizationId: authContext.organizationId,
+              rangeInDays: limit,
+            });
+
+            await schemaGraphPruningRepo.addSchemaCheckGraphPruningIssues({
+              schemaCheckId: schemaCheckID,
+              graphPruningIssues: [...graphPruningIssues.warnings, ...graphPruningIssues.errors],
+            });
+          }
+        }
+
         // Update the overall schema check with the results
         await schemaCheckRepo.update({
```

Comment:
Please make it transactional with adding schema check pruning issues. It would be also good, if we could avoid having more business logic in the rpc method.

Reply:
Done